### DISCUSSION TITLE
Performance Issues in amd64

### DISCUSSION THREAD

**Discussion by Chandler-Kluser** [BlitterStudio/amiberry#1133] [General]
**Describe the bug**

**To Reproduce**

1. Follow the build steps

2. Run amiberry in any situation

3. Notice framerate drops

**Expected behavior**
A smooth audio and video performance

**Screenshots**
Screenshots cannot present the problem

**Desktop (please complete the following information):**

- OS:
![image](https://github.com/BlitterStudio/amiberry/assets/29960960/cae4da32-9be4-4731-8bbe-76c6b52a5fbc)

- Version:
```
$ ./amiberry --help
$VER: Amiberry 5.6.1 (2023-06-13)
Usage:
 -h                         Show this help.
 --help                     

 -f <file>                  Load a configuration file.
 --config <file>            
 --model <Amiga Model>      Amiga model to emulate, from the QuickStart options.
                            Available options are: A500, A500P, A1200, A4000, CD32 and CDTV.

 --autoload <file>          Load a WHDLoad game or .CUE CD32 image using the WHDBooter.
 --cdimage <file>           Load the CD image provided when starting emulation (for CD32).
 --statefile <file>         Load a save state file.
 -s <option>=<value>        Set one or more configuration options directly, without loading a file.
                            Edit a configuration file in order to know valid parameters and settings.

Additional options:
 -0 <disk.adf>              Insert specified ADF image into emulated floppy drive 0-3.
 -1 <disk.adf>              
 -2 <disk.adf>              
 -3 <disk.adf>              

 -diskswapper=d1.adf,d2.adf Comma-separated list of disk images to pre-load to the Disk Swapper.
 -r <kick.rom>              Load main ROM from the specified path.
 -K <kick.rom>              Load extended ROM from the specified path.
 -m VOLNAME:mount_point     Attach a volume directly to the specified mount point.
 -W DEVNAME:hardfile        Attach a hardfile with the specified device name.
 -S <value>                 Sound parameter specification.
 -R <value>                 Output framerate in frames per second.
 -i                         Enable illegal memory.
 -J <xy>                    Specify joystick 0 (x) and 1 (y). Possible values: 0/1 for joystick, M for mouse, and a/b/c.
 -w <value>                 CPU emulation speed. Possible values: 0 (Cycle Exact), -1 (Max).
 -G                         Don't show the GUI, start emulation directly.
 -n                         Enable Immediate Blits. Only available when illegal memory is not enabled.
 -v <value>                 Set Chipset. Possible values: 0 (OCS), 1 (ECS Agnus), 2 (ECS Denise), 3 (Full ECS), 4 (AGA).
 -C <value>                 Set CPU specs.
 -Z <value>                 Z3 FastRAM size, value in 1MB blocks, i.e. 2=2MB.
 -U <value>                 RTG Memory size, value in 1MB blocks, i.e. 2=2MB.
 -F <value>                 Fastmem size, value in 1MB blocks, i.e. 2=2MB.
 -b <value>                 Bogomem size, value in 256KB blocks, i.e. 2=512KB.
 -c <value>                 Size of chip memory (in number of 512 KBytes chunks).
 -I <value>                 Set keyboard layout language. Possible values: de, dk, us, se, fr, it, es.
 -O <value>                 Set graphics specs.
 -H <value>                 Color mode.
 -o <amiberry cnf>=<value>  Set Amiberry configuration parameter with value.
                            See: https://github.com/midwan/amiberry/wiki/Amiberry.conf-options

Example 1:
amiberry --model A1200 -G
This will use the A1200 default settings as found in the QuickStart panel.
Additionally, it will override 'use_gui' to 'no', so that it enters emulation directly.

Example 2:
amiberry --config conf/A500.uae --statefile savestates/game.uss -s use_gui=no
This will load the conf/A500.uae configuration file, with the save state named game.
It will override 'use_gui' to 'no', so that it enters emulation directly.
```

**Additional context**
I have used some old releases and I have not noticed any performance drop, but now I am...

Love Amiberry? Please consider supporting it:
👉  https://ko-fi.com/midwan

**Comment by solskogen** [comment]
When do you notice this?

**Comment by Chandler-Kluser** [comment]
when I switched my operating system from EndeavourOS to Arch Linux, it is important to notice that the old release was kept during the system transition so I did build it from source again to test if it was something like shared object issue or something like that.

**Comment by Chandler-Kluser** [comment]
in a clean build using Amiga1200 core with workbench I noticed this issue by opening AmigaAMP:

![image](https://github.com/BlitterStudio/amiberry/assets/29960960/fc86bf91-41d5-451a-a837-e12a7afb2129)

I don't know if it is something related to the performance...

I have also tested different sound/video configuration specs, but the problem persists (I have never noticed any performance issues in older times, by the way)

**Comment by solskogen** [comment]
What do you means by performance, then? It might look like you have a problem with the sound. Make sure that works.

**Comment by Chandler-Kluser** [comment]
it is not only the sound, the video is laggy and the emulation itself too

**Comment by solskogen** [comment]
You have to provide more and better information. It seems to be a problem with your system and not amiberry itself (amiberry is just the symptom) - moving over to discussions.

**Comment by Chandler-Kluser** [comment]
I have tested a brande new clean build with a brand new clean Amiga 1200 virtual machine with a brand new AmigaOS 3.2 and the emulation is struggling.

I have also tried to get some debug message at the console, but there was no debug output in console during the emulation.

The last setup was with Scalos and it had the same issue at all...

Tips for gathering more info from amiberry emulation are welcome

**Comment by solskogen** [comment]
My best bet would be that your system is running at a lower speed than it really should, due to some power saving options. Other than that, your description is to vague. What do you mean by "struggling"? Keep in mind that amd64/x86_64 doesn't support JIT. 
What does sysinfo say about MIPS for instance?

**Reply by Chandler-Kluser** [reply]
I rebuilt my mesa drivers and the issue has been solved!!
